```python
!conda list -n STAR
```

    # packages in environment at /home/haley/.conda/envs/STAR:
    #
    # Name                    Version                   Build  Channel
    _libgcc_mutex             0.1                 conda_forge    conda-forge
    _openmp_mutex             4.5                       1_gnu    conda-forge
    attrs                     21.2.0             pyhd8ed1ab_0    conda-forge
    backports                 1.0                        py_2    conda-forge
    backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
    bamtools                  2.5.1                h9a82719_9    bioconda
    bbmap                     38.91                he522d1c_1    bioconda
    bedtools                  2.30.0               h7d7f7ad_2    bioconda
    bleach                    1.4.2                    py36_0    bioconda
    brewer2mpl                1.4.1                      py_3    conda-forge
    bzip2                     1.0.8                h7f98852_4    conda-forge
    c-ares                    1.18.1               h7f98852_0    conda-forge
    ca-certificates           2021.10.8            ha878542_0    conda-forge
    certifi                   2021.5.30        py36h5fab9bb_0    conda-forge
    click                     8.0.1            py36h5fab9bb_0    conda-forge
    colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
    coloredlogs               15.0.1           py36h5fab9bb_1    conda-forge
    colormath                 3.0.0                      py_2    conda-forge
    commonmark                0.9.1                      py_0    conda-forge
    curl                      7.80.0               h2574ce0_0    conda-forge
    cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
    dataclasses               0.7                pyhe4b4509_6    conda-forge
    decorator                 4.4.2                      py_0    conda-forge
    deeptools                 3.5.1                      py_0    bioconda
    deeptoolsintervals        0.1.9            py36hc5360cc_3    bioconda
    defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
    entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
    fastp                     0.20.1               h2e03b76_1    bioconda
    fastqc                    0.11.9               hdfd78af_1    bioconda
    font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
    fontconfig                2.13.1            hba837de_1005    conda-forge
    freetype                  2.10.4               h0708190_1    conda-forge
    future                    0.18.2           py36h5fab9bb_3    conda-forge
    ggplot                    0.6.8                    py36_0    bioconda
    html5lib                  1.1                pyh9f0ad1d_0    conda-forge
    humanfriendly             10.0             py36h5fab9bb_0    conda-forge
    icu                       67.1                 he1b5a44_0    conda-forge
    importlib-metadata        4.5.0            py36h5fab9bb_0    conda-forge
    ipykernel                 5.5.5            py36hcb3619a_0    conda-forge
    ipython                   5.8.0                    py36_1    conda-forge
    ipython_genutils          0.2.0                      py_1    conda-forge
    jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
    jsonschema                4.0.1              pyhd8ed1ab_0    conda-forge
    jupyter_client            7.1.0              pyhd8ed1ab_0    conda-forge
    jupyter_core              4.7.1            py36h5fab9bb_0    conda-forge
    kiwisolver                1.3.1            py36h605e78d_1    conda-forge
    krb5                      1.19.2               hcc1bbae_3    conda-forge
    ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
    libblas                   3.9.0           12_linux64_openblas    conda-forge
    libcblas                  3.9.0           12_linux64_openblas    conda-forge
    libcurl                   7.80.0               h2574ce0_0    conda-forge
    libdeflate                1.0                  h14c3975_1    bioconda
    libedit                   3.1.20191231         he28a2e2_2    conda-forge
    libev                     4.33                 h516909a_1    conda-forge
    libffi                    3.4.2                h7f98852_5    conda-forge
    libgcc                    7.2.0                h69d50b8_2    conda-forge
    libgcc-ng                 11.2.0              h1d223b6_11    conda-forge
    libgfortran-ng            11.2.0              h69a702a_11    conda-forge
    libgfortran5              11.2.0              h5c6108e_11    conda-forge
    libgomp                   11.2.0              h1d223b6_11    conda-forge
    libiconv                  1.16                 h516909a_0    conda-forge
    liblapack                 3.9.0           12_linux64_openblas    conda-forge
    libnghttp2                1.43.0               h812cca2_1    conda-forge
    libopenblas               0.3.18          pthreads_h8fe5266_0    conda-forge
    libpng                    1.6.37               h21135ba_2    conda-forge
    libsodium                 1.0.18               h36c2ea0_1    conda-forge
    libssh2                   1.10.0               ha56f1ee_2    conda-forge
    libstdcxx-ng              11.2.0              he4da1e4_11    conda-forge
    libuuid                   2.32.1            h7f98852_1000    conda-forge
    libxml2                   2.9.10               h68273f3_2    conda-forge
    libzlib                   1.2.11            h36c2ea0_1013    conda-forge
    lzstring                  1.0.4                   py_1001    conda-forge
    markdown                  3.3.6              pyhd8ed1ab_0    conda-forge
    markupsafe                2.0.1            py36h8f6f2f9_0    conda-forge
    matplotlib                3.2.2                         1    conda-forge
    matplotlib-base           3.2.2            py36h5fdd944_1    conda-forge
    mistune                   0.8.4           py36h8f6f2f9_1004    conda-forge
    multiqc                   1.10.1             pyhdfd78af_1    bioconda
    nbconvert                 5.6.1              pyhd8ed1ab_2    conda-forge
    nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
    ncurses                   6.2                  h58526e2_4    conda-forge
    nest-asyncio              1.5.1              pyhd8ed1ab_0    conda-forge
    networkx                  2.5.1              pyhd8ed1ab_0    conda-forge
    numpy                     1.19.5           py36hfc0c790_2    conda-forge
    openjdk                   10.0.2            h14c3975_1015    conda-forge
    openssl                   1.1.1l               h7f98852_0    conda-forge
    pandas                    1.1.5            py36h284efc9_0    conda-forge
    pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
    parallel                  20210622             ha770c72_0    conda-forge
    patsy                     0.5.2              pyhd8ed1ab_0    conda-forge
    pbzip2                    1.1.13                        0    conda-forge
    perl                      5.32.1          1_h7f98852_perl5    conda-forge
    pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
    pickleshare               0.7.5                   py_1003    conda-forge
    pip                       21.3.1             pyhd8ed1ab_0    conda-forge
    plotly                    5.4.0              pyhd8ed1ab_0    conda-forge
    prompt_toolkit            1.0.15                     py_1    conda-forge
    ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
    py2bit                    0.3.0            py36hc5360cc_5    bioconda
    pybigwig                  0.3.18           py36h0c3496a_1    bioconda
    pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
    pyparsing                 3.0.6              pyhd8ed1ab_0    conda-forge
    pyrsistent                0.17.3           py36h8f6f2f9_2    conda-forge
    pysam                     0.15.3           py36hda2845c_1    bioconda
    python                    3.6.13          hb7a2778_2_cpython    conda-forge
    python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
    python_abi                3.6                     2_cp36m    conda-forge
    pytz                      2021.3             pyhd8ed1ab_0    conda-forge
    pyyaml                    5.4.1            py36h8f6f2f9_1    conda-forge
    pyzmq                     22.3.0           py36h7068817_0    conda-forge
    readline                  8.1                  h46c0cb4_0    conda-forge
    requests                  2.12.5                   py36_0    conda-forge
    rich                      10.4.0           py36h5fab9bb_0    conda-forge
    samtools                  1.7                           1    bioconda
    scipy                     1.5.3            py36h81d768a_1    conda-forge
    setuptools                49.6.0           py36h5fab9bb_3    conda-forge
    simplegeneric             0.8.1                      py_1    conda-forge
    simplejson                3.8.1                    py36_0    bioconda
    six                       1.16.0             pyh6c4a22f_0    conda-forge
    spectra                   0.0.11                     py_1    conda-forge
    sqlite                    3.37.0               h9cd32fc_0    conda-forge
    star                      2.7.9a               h9ee0642_0    bioconda
    statsmodels               0.12.2           py36h92226af_0    conda-forge
    subread                   2.0.1                h5bf99c6_1    bioconda
    tenacity                  8.0.1              pyhd8ed1ab_0    conda-forge
    testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
    tk                        8.6.11               h27826a3_1    conda-forge
    tornado                   6.1              py36h8f6f2f9_1    conda-forge
    traitlets                 4.3.3              pyhd8ed1ab_2    conda-forge
    typing_extensions         3.10.0.2           pyha770c72_0    conda-forge
    wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
    webencodings              0.5.1                      py_1    conda-forge
    wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
    xz                        5.2.5                h516909a_1    conda-forge
    yaml                      0.2.5                h516909a_0    conda-forge
    zeromq                    4.3.4                h9c3ff4c_1    conda-forge
    zipp                      3.6.0              pyhd8ed1ab_0    conda-forge
    zlib                      1.2.11            h36c2ea0_1013    conda-forge


# Trimming & Filtering Using Fastp
Run Fastp on the samples which will extract the UMI at the start of read1, trim aligners, etc. 
will output first into rawdata, this is fixed below 


```bash
%%bash
mkdir ./fastp_output
for i in `ls -1 ./rawdata/*_001.fastq.gz | sed 's/.fastq.gz//'`
do
    #fastp 
    fastp -i $i\.fastq.gz -o $i\_fastp_output.fastq.gz \
        -U --umi_loc=read1 --umi_len=12 --thread=14 --adapter_sequence='AGATCGGAAGAGCACACGTCTGAACTCCAGTCA' -x
    mv fastp.html $i\_fastp.html
    mv fastp.json $i\_fastp.json
done  
```


```python
#move the fastp output files into a new folder (aka away from raw data) 
!mv ./rawdata/*fastp* ./fastp_output
!mkdir ./fastp_output/etcfiles
!mv ./fastp_output/*fastp.* ./fastp_output/etcfiles
#move all .gz files into data folder 
!mkdir ./fastp_output/data
!mv ./fastp_output/*.gz ./fastp_output/data
```


```bash
%%bash
#move respective experiment types into subfolders 
mkdir ./fastp_output/data/mouse
mkdir ./fastp_output/data/hypoxic_invitro
mv ./fastp_output/data/*_gnoto_* ./fastp_output/data/mouse
mv ./fastp_output/data/*hypoxic* ./fastp_output/data/hypoxic_invitro
```

# Fastqc - quality control (read #, % unique, % duplicated, quality, etc.) 


```bash
%%bash
mkdir fastQC
# mkdir fastQC/rawdata
mkdir fastQC/fastp

#very fast!!
find ./fastp_output -name "*_output.fastq.gz" | parallel fastqc -o ./fastQC/fastp/ {}
# find ./rawdata - name "*_fastq.gz" | parallel fastqc -o ./fastQC/rawdata/ {}
```


```python
#runs multiQC which will compile results from all fastqc quality control files 
!multiqc /data/haley/RNAseq/mouseexp4 -f
```

# STAR alignment to the genome 

## genome index generation
(need to be made with the version of star that you are using for the alignments (you will get an error if not) 


```bash
%%bash
#Candida albicans A22 A allele genome generation
STAR --runThreadN 10 \
    --runMode genomeGenerate \
    --genomeDir ~/STARgenomes/Ca22genome_Aallele \
    --genomeFastaFiles  ~/STARgenomes/Ca22genome_Aallele/C_albicans_SC5314_version_A22-s07-m01-r105_chromosomes_A.fasta \
    --genomeSAindexNbases 10 
```


```bash
%%bash
#concatenated Candida albicans A22 A allele-E. faecalis genome generation
STAR --runThreadN 10 \
    --runMode genomeGenerate \
    --genomeDir ~/STARgenomes/Ca22Aallele-Ef_genome \
    --genomeFastaFiles  ~/STARgenomes/Ca22Aallele-Ef_genome/Ca22_Ef_chromosomes.fa \
    --genomeSAindexNbases 10
```


```bash
%%bash
#Candida parapsilosis genome generation
STAR --runThreadN 10 \
    --runMode genomeGenerate \
    --genomeDir ~/STARgenomes/Cp_genome \
    --genomeFastaFiles  ~/STARgenomes/Cp_genome/C_parapsilosis_CDC317_version_s01-m03-r42_chromosomes.fasta \
    --sjdbGTFfile ~/STARgenomes/Cp_genome/C_parapsilosis_CDC317_version_s01-m03-r42_features.gtf \
    --genomeSAindexNbases 10
```


```bash
%%bash
#Candida parapsilosis genome generation
!STAR --runThreadN 10 \
    --runMode genomeGenerate \
    --genomeDir ./genome/CpEf_genome \
    --genomeFastaFiles ./genome/CpEf_genome/CpEf_genome.fa \
    --sjdbGTFfile ./genome/CpEf_genome/CpEf_annotations.gtf \
    --genomeSAindexNbases 10
```


```bash
%%bash
#E. faecalis genome generation
STAR --runThreadN 10 \
    --runMode genomeGenerate \
    --genomeDir ~/STARgenomes/Ef_genome \
    --genomeFastaFiles ~/STARgenomes/Ef_genome/Enterococcus_faecalis_og1rf.ASM17257v2.dna.chromosome.Chromosome.fa \
    --sjdbGTFfile ~/STARgenomes/Ef_genome/Enterococcus_faecalis_og1rf.ASM17257v2.46.gtf \
    --sjdbOverhang 62 \
    --genomeSAindexNbases 9
```

## STAR alignment to genomes
NOTE: there are nuances in how the data files were named that allow me to call only the files I want in each loop

### Alignment of Mouse samples


```bash
%%bash 
#create directory for output for STAR and subfolders for organization
mkdir ./starout/
mkdir ./starout/mouse
mkdir ./starout/mouse/CaEf_starout

#aligns CaEf mice to CaEf genome
for i in $(ls ./fastp_output/data/mouse/*_CaEf*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Ca22Aallele-Ef_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN 20 \
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/mouse/CaEf_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 \
              --alignIntronMax 4000 --alignIntronMin 4


    echo "Completed $i"
done
```


```bash
%%bash

mkdir ./starout/mouse/Ca_starout/ 

#aligns the Ca mice to Ca genome 
for i in $(ls ./fastp_output/data/mouse/*_21*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Ca22genome_Aallele/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN 15 \
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./Ca_starout_Aallele/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 \
              --outMultimapperOrder Random --alignIntronMax 4000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```bash
%%bash
mkdir ./starout/mouse/CpEf_starout 

#aligns the CpEf mice to CpEf genome 
for i in $(ls ./fastp_output/data/mouse/*_CpEf*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/CpEf_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN 20 \
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/mouse/CpEf_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 4000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```bash
%%bash
mkdir ./starout/mouse/CpEf_starout 

#aligns the CpEf mice to CpEf genome 
for i in $(ls ./fastp_output/data/mouse/*_CpEf*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/CpEf_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN 20 \
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/mouse/CpEf_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 4000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```bash
%%bash
mkdir ./starout/mouse/Cp_starout

#aligns the Cp mice to Cp genome 
for i in $(ls ./fastp_output/Cp_mice/*_24*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Cp_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN 15 \
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./Cp_starout2/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 \
              --outMultimapperOrder Random --alignIntronMax 5000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```bash
%%bash
mkdir ./starout/mouse/Ef_starout/ 

#aligns the Ef mice to Ef genome 
for i in $(ls ./fastp_output/data/mouse/*Enterococcus_*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Ef_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN  20\
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout4/mouse/Ef_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 5000 --alignIntronMin 4
    echo "Completed $i"
  done
```

### Alignment of in vitro samples


```bash
%%bash
mkdir ./starout/invitro
mkdir ./starout/invitro/Ef_starout
#aligns the Ef invitro samples to Ef genome 
for i in $(ls ./fastp_output/data/hypoxic_invitro/*Enterococcus_faecalis*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Ef_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN  20\
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/invitro/Ef_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 1000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```bash
%%bash
mkdir ./starout/invitro/Ca_starout

#aligns the Ca invitro samples to Ca genome 

for i in $(ls ./fastp_output/data/hypoxic_invitro/*albicans_hypoxic*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Ca22genome_Aallele/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN  20\
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/invitro/Ca_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 5000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```bash
%%bash

mkdir ./starout/invitro/CaEf_starout

for i in $(ls ./fastp_output/data/hypoxic_invitro/*albicans_Entero*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Ca22Aallele-Ef_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN  20\
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/invitro/CaEf_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 5000 --alignIntronMin 4
  done
```


```bash
%%bash

mkdir ./starout/invitro/Cp_starout

for i in $(ls ./fastp_output/data/hypoxic_invitro/*parapsilosis_CLIB214_hypoxic*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/Cp_genome \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN  20\
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/invitro/Cp_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 5000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```bash
%%bash 
mkdir ./starout/invitro/CpEf_starout

for i in $(ls ./fastp_output/data/hypoxic_invitro/*parapsilosis_CLIB214_Enterococcus*)
do
    b=$(basename ${i})
    STAR --genomeDir ~/STARgenomes/CpEf_genome/ \
              --runMode alignReads \
              --readFilesIn $i \
              --runThreadN  20\
              --readFilesCommand gunzip -c \
              --outSAMtype BAM SortedByCoordinate\
              --outReadsUnmapped Fastx \
              --outFileNamePrefix ./starout/invitro/CpEf_starout/$b\_ \
              --limitBAMsortRAM 31000000000 \
              --outSAMmultNmax 10 --alignIntronMax 5000 --alignIntronMin 4
    echo "Completed $i"
  done
```


```python
#for overview of Star results, use MultiQC 
!multiqc ./starout
```

# Extracting read counts for each species/sample using Rsubread FeatureCounts


```bash
%%bash
Calb=$(find ./starout/mouse/Ca_starout -type f -name "*.bam")
Calb+=" "
Calb+=$(find ./starout/mouse/CaEf_starout -type f -name "*.bam")
Calb+=" "
Calb+=$(find ./starout/invitro/Ca_starout -type f -name "*.bam")
Calb+=" "
Calb+=$(find ./starout/invitro/CaEf_starout -type f -name "*.bam")

echo $Calb
mkdir ./featurecounts

#extracts mRNA reads
featureCounts -T 20 -t mRNA -g Name -s 1 -a ~/STARgenomes/Ca22genome_Aallele/C_albicans_SC5314_version_A22-s07-m01-r105_features_Aallele_tuch.gff.txt -o ./featurecounts/C_alb_mRNA_STAR_counts.txt $Calb

#extracts rRNA and tRNA for informational purposes to see contamination of those read types, not necessary for downstream analysis
# featureCounts -T 20 -t rRNA -g Name -s 1 -a ~/STARgenomes/Ca22genome_Aallele/C_albicans_SC5314_version_A22-s07-m01-r105_features_Aallele_tuch.gff.txt -o ./featurecounts/C_alb_rRNA_STAR_counts.txt $Calb
# featureCounts -T 20 -t tRNA -g Name -s 1 -a ~/STARgenomes/Ca22genome_Aallele/C_albicans_SC5314_version_A22-s07-m01-r105_features_Aallele_tuch.gff.txt -o ./featurecounts/C_alb_tRNA_STAR_counts.txt $Calb
```


```bash
%%bash
Cpar=$(find ./starout/mouse/Cp_starout -type f -name "*.bam")
Cpar+=" "
Cpar+=$(find ./starout/mouse/CpEf_starout -type f -name "*.bam")
Cpar+=" "
Cpar+=$(find ./starout/invitro/Cp_starout -type f -name "*.bam")
Cpar+=" "
Cpar+=$(find ./starout/invitro/CpEf_starout -type f -name "*.bam")

echo $Cpar
mkdir ./featurecounts

#extracts mRNA reads
featureCounts -T 20 -t mRNA -g Name -s 1 -a ~/STARgenomes/Cp_genome/C_parapsilosis_CDC317_version_s01-m03-r42_features3.gff -o ./featurecounts/CpSTAR_mRNA_counts.txt $Cpar

#extracts rRNA and tRNA for informational purposes to see contamination of those read types, not necessary for downstream analysis
# featureCounts -T 20 -t rRNA -g Name -s 1 -a ~/STARgenomes/Cp_genome/C_parapsilosis_CDC317_version_s01-m03-r42_features3.gff -o ./featurecounts/CpSTAR_rRNA_counts.txt $Cpar
# featureCounts -T 20 -t tRNA -g Name -s 1 -a ~/STARgenomes/Cp_genome/C_parapsilosis_CDC317_version_s01-m03-r42_features3.gff -o ./featurecounts/CpSTAR_tRNA_counts.txt $Cpar
```


```bash
%%bash
Ef=$(find ./starout/mouse/Ef_starout -type f -name "*.bam")
Ef+=" "
Ef+=$(find ./starout/mouse/CaEf_starout -type f -name "*.bam")
Ef+=" "
Ef+=$(find ./starout/mouse/CpEf_starout -type f -name "*.bam")
Ef+=" "
Ef+=$(find ./starout/invitro/Ef_starout -type f -name "*.bam")
Ef+=" "
Ef+=$(find ./starout/invitro/CaEf_starout -type f -name "*.bam")
Ef+=" "
Ef+=$(find ./starout/invitro/CpEf_starout -type f -name "*.bam")


echo $Ef

featureCounts -T 10 -t gene -g ID -a ~/STARgenomes/Ef_genome/Enterococcus_faecalis_og1rf.ASM17257v2.46.chromosome.Chromosome.gff3 -o ./featurecounts/EfSTAR_mRNA_counts.txt $Ef
```


```python

```
